src/Microsoft.Extensions.ProjectModel.Sources/MsBuild/MsBuildProjectDependencyProvider.cs

```diff
@@ -0,0 +1,155 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using Microsoft.Build.Execution;
+using Microsoft.Extensions.ProjectModel.Resolution;
+
+namespace Microsoft.Extensions.ProjectModel
+{
+    internal class MsBuildProjectDependencyProvider
+    {
+        private const string PackageDependencyItemType = "_DependenciesDesignTime";
+        private const string ResolvedReferenceItemType = "ReferencePath";
+        private const string ItemSpecKey = "OriginalItemSpec";
+        private const string NugetSourceTypeKey = "NugetSourceType";
+        private const string VersionKey = "Version";
+
+        private readonly ProjectInstance _projectInstance;
+        public MsBuildProjectDependencyProvider(ProjectInstance projectInstance)
+        {
+            if (projectInstance == null)
+            {
+                throw new ArgumentNullException(nameof(projectInstance));
+            }
+            _projectInstance = projectInstance;
+        }
+
+        public IEnumerable<DependencyDescription> GetPackageDependencies()
+        {
+            var packageItems = _projectInstance.GetItems(PackageDependencyItemType);
+            var packageInfo = new Dictionary<string, DependencyDescription>();
+            if (packageItems != null)
+            {
+                foreach (var packageItem in packageItems)
+                {
+                    var packageDependency = CreateDependencyDescriptionFromItem(packageItem);
+                    if (packageDependency != null)
+                    {
+                        packageInfo[packageItem.EvaluatedInclude] = packageDependency;
+                    }
+                }
+
+                // 2nd pass to populate dependencies;
+
+                PopulateDependencies(packageInfo, packageItems);
```

Comment:
Every item here is represented using an 'itemspec' ( `item.EvaluatedInclude`)
This itemSpec is something like this (for a package)
`"TFMName/PackageName/PackageVersion"`

Lets say `Package A` has a dependency on `Package B`.
The way this is represented is 
The item representing `Package A` would have metadata with key as `Dependencies`
And the value for this metadata will be `"PackageName/PackageVersion"`. (as a `;` delimited string in case of multiple dependencies)

So to get more information regarding this, we need to lookup the item with itemspec 
`"TFMName/"` + whatever we got by parsing the Dependencies Metadata.